pkg/application/options.go

```diff
@@ -109,72 +108,33 @@ func WithConfigFile(filePath string, fileType string) Option {
 	}
 }
 
-func Flag(flag string, setting string, def any, usage string) func() (string, string, any, string) {
-	return func() (string, string, any, string) {
-		return flag, setting, def, usage
-	}
-}
-
-func WithConfigFlags(args []string, flags ...func() (string, string, any, string)) Option {
+func WithConfigFlags(args []string, opts any) Option {
 	args = funk.Filter(args, func(s string) bool {
 		return strings.HasPrefix(s, "-")
 	})
 
 	return func(app *App) {
 		app.addConfigOption(func(config cfg.GosoConf) error {
-			settings := make(map[string]any)
-			names := make([]string, 0)
-
-			fs := flag.NewFlagSet("fs", flag.ContinueOnError)
-
-			for _, f := range flags {
-				name, key, def, usage := f()
-				names = append(names, name)
-
-				switch d := def.(type) {
-				case bool:
-					settings[key] = &d
-					fs.BoolVar(&d, name, d, usage)
-				case float64:
-					settings[key] = &d
-					fs.Float64Var(&d, name, d, usage)
-				case int:
-					settings[key] = &d
-					fs.IntVar(&d, name, d, usage)
-				case int64:
-					settings[key] = &d
-					fs.Int64Var(&d, name, d, usage)
-				case string:
-					settings[key] = &d
-					fs.StringVar(&d, name, d, usage)
-				case time.Duration:
-					settings[key] = &d
-					fs.DurationVar(&d, name, d, usage)
-				default:
-					return fmt.Errorf("unknown flag type %T for arg %s", d, name)
-				}
+			if _, err := flags.NewParser(opts, flags.IgnoreUnknown).ParseArgs(args); err != nil {
```

Comment:
Do we still need the filter for args to start with `-`? I remember that you could do `--foo=bar` as well as `--foo bar` with go-flags, so I guess the filter would actually break that

Reply:
oh yes, that filter can be removed now..